### PR TITLE
Windows CI + misc ci improvements

### PR DESCRIPTION
**Pull request by wmmc88**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -20,7 +23,7 @@ jobs:
       fail-fast: false # We want all the linters to run even if one fails
       matrix:
         linter: [flake8, clang-format, xmllint]
-        # , clang-format, xmllint, cppcheck
+        # , clang-format, xmllint, cppcheck TODO FIX
```
Comment (wmmc88) [inline .github/workflows/ci.yaml:26]: resolve before merge

**Comment by wmmc88** [comment]
changes here are required for #171 to merge

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -9,22 +9,21 @@ on:
   schedule: # Trigger a job on master at midnight UTC every day
     - cron: '0 0 * * *'
 
+env: 
+  ROS_DISTRO: foxy
+  PACKAGES_TO_TEST: "uwrt_mars_rover"
```
Comment (wmmc88) [inline .github/workflows/ci.yaml:14]: All future packages must be added to this variable in order to be tested by CI

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -9,22 +9,21 @@ on:
...
+  ROS_DISTRO: galactic
+  PACKAGES_TO_TEST: "uwrt_mars_rover"
+
 jobs:
   linters:
     name: ament-${{ matrix.linter }}
     runs-on: ubuntu-latest
     container:
-      image: rostooling/setup-ros-docker:ubuntu-focal-ros-foxy-ros-base-latest
+      image: rostooling/setup-ros-docker:ubuntu-focal-ros-galactic-desktop-latest
```
Comment (wmmc88) [inline .github/workflows/ci.yaml:21]: todo: try to make this use `env.ros_distro`

**Review by niiquaye — APPROVED** [review]

**Pull request merged**